crates/aptos-api-tester/src/main.rs

```diff
@@ -46,77 +44,63 @@ async fn handle_result<Fut: Future<Output = Result<(), TestFailure>>>(
+        &network_name.to_string(),
         time,
     );
     info!(
         "{} {} result:{:?} in time:{:?}",
-        network_type.to_string(),
+        network_name.to_string(),
         test_name.to_string(),
         output,
         time,
     );
-
-    Ok(output)
 }
 
-async fn test_flows(
-    network_type: NetworkName,
-    client: Client,
-    faucet_client: FaucetClient,
-) -> Result<()> {
-    info!("testing {}", network_type.to_string());
-
-    // create clients
-    let coin_client = CoinClient::new(&client);
-    let token_client = TokenClient::new(&client);
-
-    // create and fund account for tests
-    let mut giray = LocalAccount::generate(&mut rand::rngs::OsRng);
-    faucet_client.fund(giray.address(), 100_000_000).await?;
-    info!("{:?}", giray.address());
-
-    let mut giray2 = LocalAccount::generate(&mut rand::rngs::OsRng);
-    faucet_client.fund(giray2.address(), 100_000_000).await?;
-    info!("{:?}", giray2.address());
+async fn test_flows(network_name: NetworkName) -> Result<()> {
+    info!("testing {}", network_name.to_string());
 
     // Test new account creation and funding
-    // this test is critical to pass for the next tests
-    match handle_result(
-        TestName::NewAccount,
-        network_type,
-        test_newaccount(&client, &giray, 100_000_000),
-    )
-    .await?
-    {
-        TestResult::Success => {},
-        _ => return Err(anyhow!("returning early because new account test failed")),
-    }
+    let handle_newaccount = tokio::spawn(async move {
+        process_result(
+            TestName::NewAccount,
+            network_name,
+            wrapper_newaccount(network_name),
+        )
+        .await;
+    });
 
     // Flow 1: Coin transfer
-    let _ = handle_result(
-        TestName::CoinTransfer,
-        network_type,
-        test_cointransfer(&client, &coin_client, &mut giray, giray2.address(), 1_000),
-    )
-    .await;
+    let handle_cointransfer = tokio::spawn(async move {
+        process_result(
+            TestName::CoinTransfer,
+            network_name,
+            wrapper_cointransfer(network_name),
+        )
+        .await;
+    });
 
     // Flow 2: NFT transfer
-    let _ = handle_result(
-        TestName::NftTransfer,
-        network_type,
-        test_mintnft(&client, &token_client, &mut giray, &mut giray2),
-    )
-    .await;
+    let handle_nfttransfer = tokio::spawn(async move {
+        process_result(
+            TestName::NftTransfer,
+            network_name,
+            wrapper_nfttransfer(network_name),
+        )
+        .await;
+    });
 
     // Flow 3: Publishing module
-    let _ = handle_result(
+    process_result(
         TestName::PublishModule,
-        network_type,
-        test_module(&client, &mut giray),
+        network_name,
+        wrapper_publishmodule(network_name),
     )
     .await;
 
+    handle_newaccount.await?;
+    handle_cointransfer.await?;
+    handle_nfttransfer.await?;
```

Comment:
You can also do a join_all, which I believe will ensure they all happen in parallel. 
I think this might spawn them in tasks, but run them in serial currently.  How do the start times look? https://docs.rs/futures/latest/futures/future/fn.join_all.html

Reply:
ok changed this but no effect because we still await for publish_module before we await for the handles and publish_module finishes last